ntcore/src/main/native/cpp/InstanceImpl.cpp

```diff
@@ -113,12 +113,15 @@ void InstanceImpl::StartServer(std::string_view persistFilename,
 }
 
 void InstanceImpl::StopServer() {
-  std::scoped_lock lock{m_mutex};
-  if ((networkMode & NT_NET_MODE_SERVER) == 0) {
-    return;
+  std::shared_ptr<NetworkServer> server;
+  {
+    std::scoped_lock lock{m_mutex};
+    if ((networkMode & NT_NET_MODE_SERVER) == 0) {
+      return;
+    }
+    server = std::move(m_networkServer);
```

Comment:
Do we have tests to ensure this doesn't result in a use-after-move?

Reply:
No need. Std::shared_ptr guarantees the moved-from shared_ptr is null.

Reply:
Yeah, but using an empty shared_ptr also segfaults... (or has other undesired results)

Reply:
Previously this code did m_networkServer.reset(), which also set m_networkServer to null.